### PR TITLE
@types/react-native update image prop types

### PR DESCRIPTION
**Pull request by gyzerok**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://facebook.github.io/react-native/docs/image.html

### CODE REVIEW

**Comment by gyzerok** [comment]
The changes are a bit big since I switched order of `ImageURISource` and `ImagePropertiesIOS` so related types are closer to each other.

File: react-native/index.d.ts [hunk 1]
```diff
@@ -3270,6 +3281,11 @@ export interface ImageProperties extends ImagePropertiesIOS, React.Props<Image>
     onLayout?: (event: LayoutChangeEvent) => void;
 
     /**
+     * Invoked on load error with {nativeEvent: {error}}
+     */
+    onError?: (error: { nativeEvent: any }) => void
```
Comment (gyzerok) [inline react-native/index.d.ts]: This is not specific to iOS only, so been moved to general prop type.

File: react-native/index.d.ts [hunk 2]
```diff
@@ -3161,6 +3161,51 @@ export interface ImageStyle extends FlexStyle, TransformsStyle, ShadowStyleIOS {
...
+     * these will be used to set the default `<Image/>` component dimensions.
+     */
+    width?: number,
+    height?: number,
+    /**
+     * `scale` is used to indicate the scale factor of the image. Defaults to 1.0 if
+     * unspecified, meaning that one image pixel equates to one display point / DIP.
+     */
+    scale?: number,
+}
```
Comment (alloy) [inline react-native/index.d.ts]: Should the `cache` property be included as well? https://github.com/facebook/react-native/blob/7c9173bc9ef4683d2fe82ad1fcbe81b44c635f99/Libraries/Image/ImageSourcePropType.js#L45-L70

**Review by alloy — COMMENTED** [review]
LGTM 👍  Possibly just the addition of the `cache` property.

File: react-native/index.d.ts (hunk 2 above)
Comment (gyzerok) [inline react-native/index.d.ts]: Yeah, sure, thanks for suggestion!

**Comment by gyzerok** [comment]
@paulvanbrenk it seems like everything is good. Can I request merging this, please?

**Pull request merged**